tic_tac_toe/game.py

```diff
@@ -87,18 +125,57 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    
+    #If the board is full, or we have a winner already
+    if _board_is_full(game["board"]) or game["winner"]:
+        raise InvalidMovement("Game is over.")
+        
+    #If it's not this player's turn
+    if get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next.'.format(get_next_turn(game)))
+    
+    #If the position doesn't exist   
+    if not _position_is_valid(position):
+        raise InvalidMovement("Position out of range.")
+    
+    if not _position_is_empty_in_board(position, game["board"]):
+        raise InvalidMovement("Position already taken.")
+    
+    #Otherwise update the board    
+    game["board"][position[0]][position[1]] = player #Update the board with the new move
+    if player == game["player1"]:
+        game["next_turn"] = game["player2"]
+    else:
+        game["next_turn"] = game["player1"]
+            
+    
+    #If player wins, update the winner and raise GameOver
+    if _check_winning_combinations(game["board"], player):
+       # game["next_turn"] = None
+        game["winner"] = player
+        raise GameOver('"' + player + '" wins!')
+        
+    #If the board is full after the move, it's a tie
+    if _board_is_full(game["board"]):
+        # game["next_turn"] = None
+        game["winner"] = None
+        raise GameOver("Game is tied!")
+
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    return "\n"+("\n--------------\n".join([
+        "  |  ".join([cell for cell in row ])
+        for row in game["board"]
+    ]))+"\n"
 
 
 def get_next_turn(game):
     """
     Returns the player who plays next, or None if the game is already over.
     """
-    pass
+    return game["next_turn"] if not game["winner"] else None
+
```

Comment:
Really good code guys! Lots of good ideas our group could learn from. Other than the bug I pointed out above, I would say my only other comment would be that your range(x) functions don't require two arguments since the 0 is implied. I would also say that your variable assignment (such as with board = the game board) is unneeded and takes up unnecessary resources (if I'm not mistaken), but that's really tiny stuff that won't really improve your code very much. Good job!